src/errors/kinds.rs

```diff
@@ -184,6 +184,8 @@ pub enum ErrorKind {
     FloatType,
     #[strum(message = "Input should be a valid number, unable to parse string as an number")]
     FloatParsing,
+    #[strum(message = "Input should be finite (neither infinite nor NaN)")]
```

Comment:
```suggestion
    #[strum(message = "Input should be a finite number")]
```
A bit less explicit but clearly "finite" means "not infinite" and "number" means "not NaN"?